### DISCUSSION TITLE
Embedding files in a generated PDF?

### DISCUSSION THREAD

**Discussion by ToxicFrog** [tectonic-typesetting/tectonic#1020] [Q&A]
Normally I'd do this with the `embedfile` or `embedall` packages, e.g.
```
\documentclass[10pt,letterpaper,sans]{article}
\usepackage{embedfile}
\embedfile{test.tex}
\begin{document} testing \end{document}
```
But those require pdflatex, which is not what Tectonic uses. So I tried Navigator, which says it supports xdvipdfmx:
```
\documentclass[10pt,letterpaper,sans]{article}
\usepackage{navigator}
\embeddedfile{source}{test.tex}
\begin{document} testing \end{document}
```
and this compiles, and the resulting PDF even claims it has an embedded file named "test.tex", but opening the embedded file shows it to be empty, and the build log includes the following output:
```
$ tectonic test.tex
Running TeX ...
Rerunning TeX because "test.aux" changed ...
Running xdvipdfmx ...
warning: File "test.tex" not found.
warning: Interpreting special command fstream (pdf:) failed.
warning: >> at page="1" position="(133.768, 667.198)" (in PDF)
warning: >> xxx "pdf: fstream @nav_nav@embeddedfile:source (test.tex) << /Typ..."
warning: >> Reading special command stopped around >> << /Type/EmbeddedFile >><<
warning: Object @nav_nav@embeddedfile:source used, but not defined. Replaced by null.
Writing `test.pdf` (2.96 KiB)
Skipped writing 1 intermediate files (use --keep-intermediates to keep them)
```
At first I thought it might be running that stage of the build in a different pwd or something, so that the relative path to the source wasn't resolving properly, but `strace` has ruled that out. So what do I need to do to get navigator embedded files working?

I have a workaround in the meantime running `pdfattach` after the build to attach the source files, but it would be nice to not have to do that.

**Comment by vlasakm** [comment]
Looks like this issue: https://github.com/tectonic-typesetting/tectonic/issues/935#issuecomment-1279770087, which we saw with the `attachfile2` package.

The fix is in [0.12.0](https://github.com/tectonic-typesetting/tectonic/blob/release/CHANGELOG.md#tectonic-0120-2022-10-27). Are you using an older version of Tectonic by chance?
